src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -24,7 +24,7 @@ public static bool IsInvalid(this IOperation operation, Compilation compilation,
 
             // if wrong compilation is given, GetSemanticModel will throw due to tree not belong to the given compilation.
             var model = compilation.GetSemanticModel(operation.Syntax.SyntaxTree);
-            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.Severity == DiagnosticSeverity.Error);
+            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.DefaultSeverity == DiagnosticSeverity.Error);
```

Comment:
Severity can change, but DefaultSeverity can't. for example, warning can become error if error on warning is on. and we don't want IsInvalid to return different value on options.

tagging @genlu